endpoints/remotehosts/remotehosts.py

```diff
@@ -2020,12 +2052,24 @@ def remote_image_manager(thread_name, remote_name, connection, image_max_cache_s
             thread_logger(thread_name, "Podman image '%s' is not present in rickshaw container image census, removing it from the image cache" % (image), remote_name = remote_name, log_prefix = log_prefix)
             deletes["podman"].append(image)
             remove_image(thread_name, remote_name, log_prefix, connection, image)
-        elif images["podman"][image]["created"] < image_expiration:
-            thread_logger(thread_name, "Podman image '%s' has expired, removing it from the image cache" % (image), remote_name = remote_name, log_prefix = log_prefix)
+
+        if image in images["quay"]:
+            if images["quay"][image]["end_ts"] < image_expiration:
+                thread_logger(thread_name, "Podman image '%s' has been evaluated based on it's expiration data and has expired, removing it from the image cache" % (image), remote_name = remote_name, log_prefix = log_prefix)
+                deletes["podman"].append(image)
+                deletes["rickshaw"].append(image)
+                remove_image(thread_name, remote_name, log_prefix, connection, image)
+            else:
+                thread_logger(thread_name, "Podman image '%s' has been evaluated based on it's expiration data and has not expired" % (image), remote_name = remote_name, log_prefix = log_prefix)
+        elif images["podman"][image]["created"] < image_created_expiration:
```

Comment:
Are you sure images["podman"][image] exists ? This is a 'elif' for  if image in images["quay"]:, so not having  if image in images["quay"] implies in having  if image in images["podman"] ?

Reply:
The loop is `for image in images["podman"].keys():`.